nf_core/components/list.py

```diff
@@ -11,11 +12,11 @@
 
 
 class ComponentList(ComponentCommand):
-    def __init__(self, component_type, pipeline_dir, remote=True, remote_url=None, branch=None, no_pull=False):
+    def __init__(self, component_type, pipeline_dir, remote=True, remote_url=None, branch=None, no_pull=False) -> None:
```

Comment:
```suggestion
    def __init__(self, component_type, pipeline_dir, remote:bool=True, remote_url:str=None, branch:str=None, no_pull:bool=False) -> None:
```
Any reason why you didn't add these types?

Reply:
I would also add types for `component_type` and specially `pipeline_dir`, no? `pipeline_dir` can be provided by command line, so this could be one of the more useful ones.